qiskit/circuit/twirling.py

```diff
@@ -0,0 +1,53 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The twirling module."""
+
+from __future__ import annotations
+
+from qiskit._accelerate.twirling import twirl_circuit as twirl_rs
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.library.standard_gates.x import CXGate
+from qiskit.exceptions import QiskitError
+
+
+def twirl_circuit(
+    circuit: QuantumCircuit,
+    twirling_gate: Gate = CXGate,
```

Comment:
```suggestion
    twirling_gate: type[Gate] = CXGate,
```
... otherwise this would ask for `Gate` instances, I think, no? 🙂